### PR TITLE
Optimizing routes

### PR DESCRIPTION
**Pull request by notwillcasey**
- created download photo route
- optimized schema with indexing to improve read queries
- created more consistent mock data and mock data file
- refactored testing to only impact 1 mock conversation

### CODE REVIEW

File: server.test.js [hunk 1]
```diff
@@ -6,13 +6,13 @@ const request = supertest(app);
...
   it('should respond with conversation data for valid chatId', async () => {
-    const response = await request.get('/conversation?chatId=1');
+    const response = await request.get('/conversation?chatId=12');
     expect(response.status).toBe(200);
     expect(Array.isArray(response.body)).toBe(true);
   });
 
   it('should respond with a 400 status code for an invalid chatId', async () => {
-    const response = await request.get('/conversation?chatId=999');
+    const response = await request.get('/conversation?chatId=999999');
```
Comment (callumreid) [inline server.test.js:15]: Presume this is just the planning for all the users we will have

File: api/controllers.js [hunk 2]
```diff
@@ -50,6 +50,7 @@ module.exports = {
         if (!response || response.rowCount === 0) {
           res.status(400).send('UNABLE TO GET CONVERSATION - try again later');
         } else if (response === 'noID') {
+          console.log('right here');
```
Comment (callumreid) [inline api/controllers.js:53]: Somewhat confusing of a log but I trust you know what it means

File: api/controllers.js [hunk 3]
```diff
@@ -128,6 +129,20 @@ module.exports = {
         console.log(err);
         res.status(400).send(err);
       });
+  },
+
+  downloadPhoto: (req, res) => {
+    if (!req.query.url) {
+      res.status(400).send('MISSING INPUT - url is required');
```
Comment (callumreid) [inline api/controllers.js:136]: Appreciate the specific error responses

File: api/helpers.js [hunk 4]
```diff
@@ -29,6 +29,31 @@ module.exports = {
...
+      });
+    });
+  },
+
+  downloadPhoto: (url) => {
+    return new Promise((resolve, reject) => {
+      if (!url) {
+        resolve(null);
+      }
+      const s3 = new Aws.S3({
```
Comment (callumreid) [inline api/helpers.js:47]: This is some syntax I haven't seen before-- interesting to see S3 in action

File: service/utils/mockData.sql [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  (15, 1, 5, 0, 'f', 1),
+  (16, 1, 6, 0, 'f', 1);
+
+INSERT INTO
+  conversation (chatId, senderId, body, photoUrl, read)
+VALUES
+  (12, 1, 'barking meow, I just stubbed my paw', null, 't'),
+  (12, 2, 'Language, please...', null, 't'),
+  (13, 1, 'Have you heard from Mr. Meowgi??', null, 't'),
+  (13, 3, 'Not since she went to the groomers...', null, 't'),
```
Comment (callumreid) [inline service/utils/mockData.sql:16]: 😂

File: service/utils/schema.sql [hunk 6]
```diff
@@ -32,4 +33,8 @@ CREATE TABLE conversation(
 -- CSV HEADER;
 
 CREATE UNIQUE INDEX cid ON chatlist(chatId);
--- CREATE UNIQUE INDEX chid ON conversation(chatId);
\ No newline at end of file
+CREATE INDEX user1 ON chatlist(uid1);
```
Comment (callumreid) [inline service/utils/schema.sql:36]: Seem like the right choices for indexes

**Review by callumreid — APPROVED** [review]
Everything looks great here. Love the data you loaded into the db

File: api/controllers.js (hunk 2 above)
Comment (notwillcasey) [reply api/controllers.js:53]: leftover from tracking an error -- deleted in my next commit. Thanks!

**Pull request merged**